### DISCUSSION TITLE
My setup

### DISCUSSION THREAD

**Discussion by blueskytod** [timmbogner/Farm-Data-Relay-System#177] [Q&A] [answered]
Hi , 
I use the farm data relay system with esp now for some time and it works great.
I have some Lora Module Ra-02 SX1278 433MHZ with some wemos mini esp8266 , but that set-up didn't work, any one tried that ? Or I have to try with esp32 ? 
Have a great day

**Comment by timmbogner** [comment]
Hiya,
I've got a couple of those Ra-02 modules and have tested the system on them successfully. The following is a setup that I *think* should work. I haven't tested it in a while.

One of my next tasks is actually to use these Ra-02s to test the new timekeeping functionality with different MCU architectures, including Wemos8266. Let me know if these configs don't work and I will jump in and test with you.

``` cpp
#define RADIOLIB_MODULE SX1278  // ESP8266 SX1278
#define LORA_SS    15
#define LORA_RST   4
#define LORA_DIO   5
#define LORA_BUSY RADIOLIB_NC
```

**Reply by timmbogner** [reply]
@blueskytod You also need to change the frequency parameter to "433.0". You can do this either in the global file or in the config file. I have tested this configuration successfully: (an example of setting it in the config file)
``` cpp

#define RADIOLIB_MODULE SX1278  // ESP8266 SX1278
#define LORA_SS    15
#define LORA_RST   4
#define LORA_DIO   5
#define LORA_BUSY  RADIOLIB_NC
#define LORA_FREQUENCY 433.0
```

**Comment by blueskytod** [comment]
@timmbogner i always have a 'radiolib error code2 'in serial monitor , can you show me the others connection for the SPI interface on your esp8266 (Sclk,mosi,miso ...)

**Reply by timmbogner** [reply]
Sure thing,
This configuration uses the standard SPI pins [found here](https://randomnerdtutorials.com/esp8266-pinout-reference-gpios/). Beware of the difference between, for example, "D5" and "GPIO5". In configs we are always using the "GPIO" designation for a pin, which will not match the pin number printed on the Wemos board. I will edit this post shortly with a photo of my working setup for clarity.

![IMG_20230617_083003055](https://github.com/timmbogner/Farm-Data-Relay-System/assets/64260873/3352a1de-48b3-49f4-a193-97dc0d2e9213)

**Reply by timmbogner** [reply, marked as answer]
One more thing I keep forgetting: The reset pin is optional.

**Reply by blueskytod** [reply]
Thank you very much , it was very helpful , 
just a last piece of advice about an example of flow (in node red )to control a controller node ( simple irrigation system with on swith )  
again thank you

**Reply by timmbogner** [reply]
Glad you got it working!

That's a great suggestion, thanks! As I was doing this, I added "Readmes for example sketches" to my to-do list. Those will be a great place to explain things and give Node-RED examples, especially the more intricate stuff like controllers.
